src/state/slices/portfolioSlice/utils.ts

```diff
@@ -208,71 +211,54 @@ export const accountToPortfolio: AccountToPortfolio = args => {
           }
 
           portfolio.accounts.byId[accountId].assetIds.push(token.assetId)
-
-          portfolio.accountBalances.byId[accountId] = {
-            ...portfolio.accountBalances.byId[accountId],
-            [token.assetId]: token.balance,
-          }
+          portfolio.accountBalances.byId[accountId][token.assetId] = token.balance
         })
+
         break
       }
       case CHAIN_NAMESPACE.Utxo: {
+        const utxoAccount = account as Account<UtxoChainId>
         const { balance, chainId, assetId, pubkey } = account
+
         // Since btc the pubkeys (address) are base58Check encoded, we don't want to lowercase them and put them in state
         const accountId = `${chainId}:${pubkey}`
 
-        portfolio.accountBalances.ids.push(accountId)
-
-        // initialize this
-        portfolio.accountBalances.byId[accountId] = {}
-
-        // add the balance from the top level of the account
-        portfolio.accountBalances.byId[accountId][assetId] = balance
+        const hasActivity =
+          bnOrZero(balance).gt(0) ||
+          bnOrZero(utxoAccount.chainSpecific.nextChangeAddressIndex).gt(0) ||
+          bnOrZero(utxoAccount.chainSpecific.nextReceiveAddressIndex).gt(0)
 
-        // initialize
-        if (!portfolio.accounts.byId[accountId]?.assetIds.length) {
-          portfolio.accounts.byId[accountId] = {
-            assetIds: [],
-          }
-        }
-
-        portfolio.accounts.ids = Array.from(new Set([...portfolio.accounts.ids, accountId]))
-
-        portfolio.accounts.byId[accountId].assetIds = Array.from(
-          new Set([...portfolio.accounts.byId[accountId].assetIds, assetId]),
-        )
+        portfolio.accounts.ids.push(accountId)
+        portfolio.accounts.byId[accountId] = { assetIds: [assetId], hasActivity }
+        portfolio.accountBalances.ids.push(accountId)
+        portfolio.accountBalances.byId[accountId] = { [assetId]: balance }
 
         break
       }
       case CHAIN_NAMESPACE.CosmosSdk: {
         const cosmosAccount = account as Account<KnownChainIds.CosmosMainnet>
         const { chainId, assetId } = account
         const accountId = toAccountId({ chainId, account: _xpubOrAccount })
-        portfolio.accountBalances.ids.push(accountId)
 
-        portfolio.accounts.byId[accountId] = {
-          assetIds: [],
-        }
-
-        portfolio.accounts.byId[accountId].assetIds.push(assetId)
+        const hasActivity =
+          bnOrZero(cosmosAccount.chainSpecific.sequence).gt(0) ||
```

Comment:
Could also check for `cosmosAccount.chainSpecific.assets` if we feel like it but sequence should catch it + CosmosHub tokens are relatively edge case

Reply:
Have that in the loop below (L.253), figured I would leverage that instead of adding another loop